compass/job/__init__.py

```diff
@@ -0,0 +1,83 @@
+from jinja2 import Template
+from importlib import resources
+import os
+import numpy as np
+
+
+def write_job_script(config, machine, target_cores, min_cores, work_dir):
+    """
+
+    Parameters
+    ----------
+    config : compass.config.CompassConfigParser
+        Configuration options for this test case, a combination of user configs
+        and the defaults for the machine and MPAS core
+
+    machine : {str, None}
+        The name of the machine
+
+    target_cores : int
+        The target number of cores for the job to use
+
+    min_cores : int
+        The minimum number of cores for the job to use
+
+    work_dir : str
+        The work directory where the job script should be written
+    """
+
+    if config.has_option('parallel', 'account'):
+        account = config.get('parallel', 'account')
+    else:
+        account = ''
+
+    cores_per_node = config.getint('parallel', 'cores_per_node')
+
+    # as a rule of thumb, let's do the geometric mean between min and target
+    cores = np.sqrt(target_cores*min_cores)
+    nodes = int(np.ceil(cores/cores_per_node))
```

Comment:
The geometric mean seems to be a good guess for what folks might want but I'm up for other suggestions for how to handle the node count.